### PR TITLE
AOTV: Create new subscriber on signin (#385)

### LINKED ISSUE #385: Create Subscriber when new user logs in
When a new user logs in, we want to create a subscriber instance to store in the datastore.

### PR DESCRIPTION
**Pull request by ao-david**
When a new user signs in, check if there is an existing subscriber with the given email otherwise create a new subscriber.

closes #385

### CODE REVIEW

File: cmd/ausoceantv/auth.go [hunk 1]
```diff
@@ -44,7 +48,35 @@ func (svc *service) logoutHandler(c *fiber.Ctx) error {
...
+	p, err := svc.GetProfile(c)
+	if errors.Is(err, SessionNotFound) || errors.Is(err, TokenNotFound) {
+		return fiber.ErrUnauthorized
+	} else if err != nil {
+		return fmt.Errorf("unable to get profile: %w", err)
+	}
+
+	ctx := context.Background()
+
+	ID := model.NewSubscriberID(ctx, svc.settingsStore)
```
Comment (trekhopton) [inline cmd/ausoceantv/auth.go:66]: This will look different if ID is generated later by CreateSubscriber.

**Review by trekhopton — APPROVED** [review]

File: cmd/ausoceantv/auth.go [hunk 2]
```diff
@@ -44,7 +47,33 @@ func (svc *service) logoutHandler(c *fiber.Ctx) error {
 
 // callbackHandler handles callbacks from google's oauth2 flow.
 func (svc *service) callbackHandler(c *fiber.Ctx) error {
-	return svc.CallbackHandler(c)
+	err := svc.CallbackHandler(c)
+	if err != nil {
+		return err
```
Comment (saxon-milton) [inline cmd/ausoceantv/auth.go:52]: wrap error ?

File: cmd/ausoceantv/auth.go [hunk 3]
```diff
@@ -44,7 +47,33 @@ func (svc *service) logoutHandler(c *fiber.Ctx) error {
...
+	}
+
+	// Check if a user already exists.
+	p, err := svc.GetProfile(c)
+	if errors.Is(err, SessionNotFound) || errors.Is(err, TokenNotFound) {
```
Comment (saxon-milton) [inline cmd/ausoceantv/auth.go:57]: Would it be useful to have separate cases for these and then you can wrap the fiber error with more information i.e. whether it was a ```SessionNotFound``` or a ```TokenNotFound``` ?

Also, why exactly are we using the fiber error ?

File: cmd/ausoceantv/auth.go [hunk 4]
```diff
@@ -44,7 +47,33 @@ func (svc *service) logoutHandler(c *fiber.Ctx) error {
 
 // callbackHandler handles callbacks from google's oauth2 flow.
 func (svc *service) callbackHandler(c *fiber.Ctx) error {
```
Comment (saxon-milton) [inline cmd/ausoceantv/auth.go:49]: I think the fiber context should be named like fCtx or something.

File: cmd/ausoceantv/auth.go [hunk 5]
```diff
@@ -44,7 +47,33 @@ func (svc *service) logoutHandler(c *fiber.Ctx) error {
...
+
+	// Check if a user already exists.
+	p, err := svc.GetProfile(c)
+	if errors.Is(err, SessionNotFound) || errors.Is(err, TokenNotFound) {
+		return fiber.ErrUnauthorized
+	} else if err != nil {
+		return fmt.Errorf("unable to get profile: %w", err)
+	}
+
+	ctx := context.Background()
```
Comment (saxon-milton) [inline cmd/ausoceantv/auth.go:63]: So are there no ```context.Context``` in scope where ```callbackHandler``` is called that we could pass to this ?

**Review by saxon-milton — CHANGES_REQUESTED** [review]

**Pull request merged**